index.js

```diff
@@ -159,4 +159,51 @@ mung.headersAsync = function headersAsync (fn) {
     }
 }
 
+mung.writeJson = function writeJson (fn, options = {}) {
+    return function (req, res, next) {
+        const original = res.write;
+        const mungError = options.mungError;
+
+        res.write = (body, ...args) => {
+            const contentType = res.get('Content-Type');
+
+            if (res.headersSent) {
+                return;
+            }
+
+            // Do not mung on errors
+            if (!mungError && res.statusCode >= 400) {
+                return original.call(res, body);
+            }
+
+            // If response type is not application/json,
+            // just call the original res.write function
+            if (!(contentType && ~contentType.indexOf('application/json'))) {
+                return original.call(res, body);
+            }
+
+            try {
+                const json = JSON.parse(body);
+                const modifiedJSON = fn(json, req, res);
+
+                if (res.headersSent) {
+                    return;
+                }
+
+                // If null, then 204 No Content
+                if (!modifiedJSON) {
```

Comment:
`mung;.json` allows the result object to be modified in-place or a new object returned.  This only allows a  new object to be returned by the mung function.